cmd/monaco/download/download_configs_test.go

```diff
@@ -280,50 +294,90 @@ func TestDownload_Options(t *testing.T) {
-					}
-					return nil, nil
-				},
-				automationDownload: func(a client.AutomationClient, s string, automationType ...config.AutomationType) (projectv2.ConfigsPerType, error) {
-					if !tt.want.automation {
-						t.Fatalf("automation download was not meant to be called but was")
-					}
-					return nil, nil
-				},
-				bucketDownload: func(b client.BucketClient, s string) (projectv2.ConfigsPerType, error) {
-					if !tt.want.bucket {
-						t.Fatalf("automation download was not meant to be called but was")
-					}
-					return nil, nil
-				},
-				documentDownload: func(b client.DocumentClient, s string) (projectv2.ConfigsPerType, error) {
-					if !tt.want.document {
-						t.Fatalf("document download was not meant to be called but was")
-					}
-					return nil, nil
-				},
-				openPipelineDownload: func(b client.OpenPipelineClient, s string) (projectv2.ConfigsPerType, error) {
-					if !tt.want.openpipeline {
-						t.Fatalf("openpipeline download was not meant to be called but was")
-					}
-					return nil, nil
-				},
-			}
+		fn := downloadFn{
+			classicDownload: func(client.ConfigClient, string, api.APIs, classic.ContentFilters) (projectv2.ConfigsPerType, error) {
+				if !tt.want.config {
+					t.Fatalf("classic config download was not meant to be called but was")
+				}
+				return nil, nil
+			},
+			settingsDownload: func(settingsClient client.SettingsClient, s string, filters settings.Filters, settingsType ...config.SettingsType) (projectv2.ConfigsPerType, error) {
+				if !tt.want.settings {
+					t.Fatalf("settings download was not meant to be called but was")
+				}
+				return nil, nil
+			},
+			automationDownload: func(a client.AutomationClient, s string, automationType ...config.AutomationType) (projectv2.ConfigsPerType, error) {
+				if !tt.want.automation {
+					t.Fatalf("automation download was not meant to be called but was")
+				}
+				return nil, nil
+			},
+			bucketDownload: func(b client.BucketClient, s string) (projectv2.ConfigsPerType, error) {
+				if !tt.want.bucket {
+					t.Fatalf("automation download was not meant to be called but was")
+				}
+				return nil, nil
+			},
+			documentDownload: func(b client.DocumentClient, s string) (projectv2.ConfigsPerType, error) {
+				if !tt.want.document {
+					t.Fatalf("document download was not meant to be called but was")
+				}
+				return nil, nil
+			},
+			openPipelineDownload: func(b client.OpenPipelineClient, s string) (projectv2.ConfigsPerType, error) {
+				if !tt.want.openpipeline {
+					t.Fatalf("openpipeline download was not meant to be called but was")
+				}
+				return nil, nil
+			},
+			segmentDownload: func(b segment.DownloadSegmentClient, s string) (projectv2.ConfigsPerType, error) {
+				if !tt.want.segment {
+					t.Fatalf("segment download was not meant to be called but was")
+				}
+				return nil, nil
+			},
+		}
+
+		t.Run(fmt.Sprintf("%s - all temroray FF are false", tt.name), func(t *testing.T) {
```

Comment:
I am not sure what this does, but we discussed a different solution during the call. This is something completely different.
What we agreed on was the following:
1) We extend the test setup to set the environments _per test case_. We do this by allowing to set a  `map[string]string` to enable/disable ffs as required. e.g. enabling/disabling the segments FF for a single test case defined above.
2) We duplicate tests where we enable/disable FF. E.g. We have a test-case defined where segments should be downloaded, and the behavior should be tested for both if the FF is enabled/disabled. Same if we don't want to download segments. 

No new extra test runs like this here are required, and just enabling/disabling all temporary featureflags seems really odd and IMO brings no real value.

Reply:
done